pgxn/neon/control_plane_connector.c

```diff
@@ -670,14 +675,31 @@ HandleCreateRole(CreateRoleStmt *stmt)
 static void
 HandleAlterRole(AlterRoleStmt *stmt)
 {
-	const char *role_name = stmt->role->rolename;
+	char	   *role_name = NULL;
 	DefElem    *dpass;
 	ListCell   *option;
 	bool		found = false;
 	RoleEntry  *entry;
 
 	InitRoleTableIfNeeded();
 
+	switch (stmt->role->roletype)
+	{
+		case ROLESPEC_CSTRING:
+			role_name = pstrdup(stmt->role->rolename);
+			break;
+		case ROLESPEC_CURRENT_ROLE:
+		case ROLESPEC_CURRENT_USER:
+			role_name = GetUserNameFromId(GetUserId(), false);
+			break;
+		case ROLESPEC_SESSION_USER:
+			role_name = GetUserNameFromId(GetSessionUserId(), false);
+			break;
+		case ROLESPEC_PUBLIC:
+			role_name = pstrdup("public");
```

Comment:
I believe this should not happen,  because:
```
postgres=# create role public;
ERROR:  role name "public" is reserved
LINE 1: create role public;
                    ^
```
maybe throw an error in this case, just as a sanity check